src/CarsDataTable.js

```diff
@@ -1,12 +1,50 @@
 import React from 'react';
 
-const CarsDataTable = (props) => {
-  return (
-    <div>
-      The list has {props.cars.length} cars.<br />
-      <b>Your task is to render a table here</b>.
-    </div>
-  );
-};
+type Props = {
+  cars: Array<Car>
+}
+
+type Car = {
+  id: string,
+  carMake: string,
+  carModel: string,
+  carModelYear: string,
+  bodyColor: string,
+  dealerEmail: string,
+}
 
-export default CarsDataTable;
+export default class CarsDataTable extends React.Component<Props> {
+  renderRow(car) {
```

Comment:
Turning this component into a class just to be able to add this method seems a little unjustified. A more natural approach would be to create another functional component for `CarsDataTableRow`. Class components are usually only required if you  need to manage state or there is a lot of custom event handling going on.